xla/backends/profiler/gpu/device_tracer_rocm.cc

```diff
@@ -241,278 +237,129 @@ class RocmTraceCollectorImpl : public profiler::RocmTraceCollector {
-        case RocmTracerEventType::MemoryFree: {
-          if (iter != api_events_map_.end() ||
-              iter != auxiliary_api_events_map_.end()) {
-            activity_event.device_id = iter->second.device_id;
+          //MALLOC ACTIVITY, FREE ACTIVITY
+          case RocmTracerEventType::MemoryAlloc:
+          case RocmTracerEventType::MemoryFree: {
+            activity_event.device_id = api_event->second.device_id;
             aggregated_events.push_back(activity_event);
-          } else {
-            OnEventsDropped(
-                "A Malloc/Free activity was discarded."
-                " Could not find the counterpart API callback.",
-                activity_event.correlation_id);
-            DumpRocmTracerEvent(activity_event, 0, 0, ". Dropped!");
+            break;
           }
-          break;
-        }
-        /*SYNCHRONIZATION ACTIVITY*/
-        case RocmTracerEventType::Synchronization: {
-          if (iter != api_events_map_.end() ||
-              iter != auxiliary_api_events_map_.end()) {
-            // CUDA does not provide device ID for these activities.
-            // Interestingly, TF-profiler by default set the device id to 0 for
-            // CuptiTracerEvent.
-            // RocmTracerEvent type, set device by default to an unvalid
-            // device-id value. To be consistent with CUDA (in terms of having a
-            // logically valid value for device id) we update the device-id to
-            // its correct value
-            activity_event.device_id = iter->second.device_id;
+          //SYNCHRONIZATION ACTIVITY
+          case RocmTracerEventType::Synchronization: {
+            activity_event.device_id = api_event->second.device_id;
             aggregated_events.push_back(activity_event);
-          } else {
-            OnEventsDropped(
-                "A sync event activity was discarded."
-                " Could not find the counterpart API callback.",
-                activity_event.correlation_id);
-            DumpRocmTracerEvent(activity_event, 0, 0, ". Dropped!");
-          }
-          break;
-        }
-        default:
-          OnEventsDropped("Missing API-Activity information exchange. Dropped!",
+            break;
+          } 
+          default:
+            OnEventsDropped("Missing API-Activity information exchange. Dropped!",
                           activity_event.correlation_id);
-          DumpRocmTracerEvent(activity_event, 0, 0, ". Dropped!");
-          LOG(WARNING) << "A ROCm activity event with unimplemented API "
+            DumpRocmTracerEvent(activity_event, 0, 0, ". Dropped!");
+            LOG(WARNING) << "A ROCm activity event with unimplemented API "
                           "callback merge dropped! "
                           "Type="
                        << GetRocmTracerEventTypeName(activity_event.type);
-          break;
-      }  // end switch(activity_event.type)
+            break;
+        }
+      }
+    }
+
+    /*
```

Comment:
Please remove this block instead of commenting it out.

Reply:
Fixed